asllib/Lexer.mll

```diff
@@ -202,7 +202,7 @@ and token = parse
     | real_lit as lxm          { REAL_LIT(Q.of_string lxm)        }
     | '"'                      { string_lit (Buffer.create 16) lexbuf }
     | '\'' (bits as lxm) '\''  { bitvector_lit lxm                }
-    | '\'' (mask as lxm) '\''  { mask_lit lxm                     }
+    | '\'' (mask as lxm) '\''  { mask_lit lxm                     }  (* Warning: masks with no unknown 'x' characters will be lexed as bitvectors. *)
```

Comment:
I think it would be cleaner to have just the regular expression for bitmasks and lex it into some kind of token like `mask_or_bits` that will later be converted by the parser to either a bitmask or a bitvector. The parser has the right context to decide which one.

Reply:
Right. I tried it and saw the issues for myself.